### PR TITLE
Refactor `BytesLib`

### PR DESCRIPTION
**Pull request by shuhuiluo**
Reduce duplicate code and add annotation.

Mostly aesthetics. Hardhat gas snapshots show mixed results. 

`forge snapshot --diff` shows `Overall gas change: -359 (-0.007%)`.

### CODE REVIEW

File: contracts/modules/uniswap/v3/BytesLib.sol [hunk 1]
```diff
@@ -29,27 +29,48 @@ library BytesLib {
...
+    /// @return length Length of the array
+    /// @return offset Pointer to the data part of the array
+    function toLengthOffset(bytes calldata _bytes, uint256 arg)
+        internal
+        pure
+        returns (uint256 length, uint256 offset)
+    {
+        assembly {
+            // The offset of the `arg`-th element is `32 * arg`, which stores the offset of the length pointer.
+            let lengthPtr := add(_bytes.offset, calldataload(add(_bytes.offset, mul(32, arg))))
```
Comment (hensha256) [inline contracts/modules/uniswap/v3/BytesLib.sol:44]: We like to consistently use hex for our numbers in assembly - so please can you change `32` to `0x20`

**Review by hensha256 — CHANGES_REQUESTED** [review]
Thanks for making this PR! I appreciate the cleanup you've done! 🙌 Added one nit but then it looks good to go!

Sadly we cannot run our CI tests on fork PRs as they cannot access our github secrets. Please can you change the base branch of the PR to `Uniswap:byteslib-refactor`? Once we merge I will then set up a new PR of that branch into main, so that we can run our CI and check everything is okay.

File: contracts/modules/uniswap/v3/BytesLib.sol (hunk 1 above)
Comment (shuhuiluo) [reply contracts/modules/uniswap/v3/BytesLib.sol]: Sure. I intentionally used decimals as `toPool` uses decimals.

**Comment by shuhuiluo** [comment]
> Thanks for making this PR! I appreciate the cleanup you've done! 🙌 Added one nit but then it looks good to go!
> 
> Sadly we cannot run our CI tests on fork PRs as they cannot access our github secrets. Please can you change the base branch of the PR to `Uniswap:byteslib-refactor`? Once we merge I will then set up a new PR of that branch into main, so that we can run our CI and check everything is okay.

Sorry for the late response. Was busy in Denver last week. A new PR has been created. Thank you for the hard work on this elegant router.

**Pull request closed without merging**